### PR TITLE
fix autoloading in LPAP integration tests, resolves #544

### LINKED ISSUE #544: LDAP Integration Tests fail due to autoloader
For example, I want to run ``/apps/user_ldap/tests/IntegrationLib/IntegrationTestAccessGroupsMatchFilter.php`` the test will fail with:

``` 
PHP Fatal error:  Class 'OCA\User_LDAP\Tests\Integration\AbstractIntegrationTest' not found in /srv/http/nextcloud/master/apps/user_ldap/tests/Integration/Lib/IntegrationTestAccessGroupsMatchFilter.php on line 31
```

I failed to figure out how the paths are constructed by the autoloader (as in have not seen where OCA stuff is registered as PSR-4 thingy):

```
[0] => /srv/http/nextcloud/master/apps/user_ldap/tests/integration/abstractintegrationtest.php
[1] => /srv/http/nextcloud/master/apps/user_ldap/lib/tests/integration/abstractintegrationtest.php
```

Anyway, to make the test run again, I need a line like

```diff
diff --git a/apps/user_ldap/tests/Integration/Lib/IntegrationTestAccessGroupsMatchFilter.php b/apps/user_ldap/tests/Integration/Lib/IntegrationTestAccessGroupsMatchFilter.php
index 4fcc833..ca89700 100644
--- a/apps/user_ldap/tests/Integration/Lib/IntegrationTestAccessGroupsMatchFilter.php
+++ b/apps/user_ldap/tests/Integration/Lib/IntegrationTestAccessGroupsMatchFilter.php
@@ -23,11 +23,12 @@
 
 namespace OCA\User_LDAP\Tests\Integration\Lib;
 
-
 use OCA\User_LDAP\Tests\Integration\AbstractIntegrationTest;
 
 require_once __DIR__  . '/../../../../../lib/base.php';
 
+\OC::$composerAutoloader->addPsr4('OCA\\User_LDAP\\Tests\\', \OC::$SERVERROOT . '/apps/user_ldap/tests/', true);
+
 class IntegrationTestAccessGroupsMatchFilter extends AbstractIntegrationTest {
 
    /**
```

in test file.

I am doubtful whether this is the best approach. Opinions, @rullzer @nickvergessen ?

### PR DESCRIPTION
**Pull request by blizzz**
as discussed, @nickvergessen @rullzer 

requires backporting

### CODE REVIEW

File: apps/user_ldap/tests/Integration/Bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+define('CLI_TEST_RUN', true);
+require_once __DIR__  . '/../../../../lib/base.php';
+require_once __DIR__ . '/setup-scripts/config.php';
```
Comment (nickvergessen) [inline apps/user_ldap/tests/Integration/Bootstrap.php]: missing NL?

**Comment by nickvergessen** [comment]
:+1:

File: apps/user_ldap/tests/Integration/Bootstrap.php (hunk 1 above)
Comment (blizzz) [inline apps/user_ldap/tests/Integration/Bootstrap.php]: *sigh* temporary replacement IDE

**Comment by MorrisJobke** [comment]
👍

**Comment by blizzz** [comment]
@karlitschek we should backport this

**Comment by karlitschek** [comment]
yes. please backport

**Comment by blizzz** [comment]
Backport to stable10 here: https://github.com/nextcloud/server/pull/580
stable9 is not affected

**Pull request merged**